rpc/resinit/resinit_linux.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// +build linux,!android
+
+package resinit
+
+// Versions of glibc prior to 2.26 (not yet released as of this writing) have a
+// longstanding bug where they don't detect changes in /etc/resolv.conf, which
+// can cause programs to get into a permanent "host not found" state even
+// though the network is up. The workaround for this bug is to call the
+// res_init() libc function to clear the cached configs, generally after a
+// failed network request.
+//
+// However, res_init() is documented *not threadsafe*. It really does crash on
+// OSX if you call it from multiple threads on e.g. macOS
+// (https://github.com/rust-lang/rust/issues/43592). Nor does putting a lock
+// around it totally let us off the hook. This is library code, after all, and
+// there's no telling what *other* libraries might be doing on parallel threads
+// in our process.
```

Comment:
The cache here is a glibc global variable, so I don't think other processes matter.